const catchError = require('../utils/catchError');
const Variable = require('../models/Variable');
const Pregunta = require('../models/Pregunta');

const getAll = catchError(async(req, res) => {
    const results = await Variable.findAll({include: [Pregunta]});
    return res.json(results);
});

const create = catchError(async(req, res) => {
    const result = await Variable.create(req.body);
    return res.status(201).json(result);
});

const getOne = catchError(async(req, res) => {
    const { id } = req.params;
    const result = await Variable.findByPk(id);
    if(!result) return res.sendStatus(404);
    return res.json(result);
});

const remove = catchError(async(req, res) => {
    const { id } = req.params;
    await Variable.destroy({ where: {id} });
    return res.sendStatus(204);
});

const update = catchError(async(req, res) => {
    const { id } = req.params;
    const result = await Variable.update(
        req.body,
        { where: {id}, returning: true }
    );
    if(result[0] === 0) return res.sendStatus(404);
    return res.json(result[1][0]);
});

module.exports = {
    getAll,
    create,
    getOne,
    remove,
    update
}